### PR TITLE
executors: Add exporter for AWS custom metric

### PR DESCRIPTION
**Pull request by eseliger**
TODO: clean up code

### CODE REVIEW

File: enterprise/cmd/frontend/internal/executorqueue/metrics/aws/reporter.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+func NewReporter(environmentLabel string) (*awsMetricReporter, error) {
+	if awsConfig.MetricNamespace == "" {
+		return nil, nil
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load aws default config")
+	}
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/executorqueue/metrics/aws/reporter.go:26]: Do we need to support another mechanism here?

File: enterprise/cmd/frontend/internal/executorqueue/metrics/gcp/reporter.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	gcpMetricType = "custom.googleapis.com/executors/queue/size"
+)
+
+func NewReporter(environmentLabel string) (*gcpMetricReporter, error) {
+	if gcpConfig.ProjectID == "" {
+		return nil, nil
+	}
+
+	log15.Info("Sending executor queue metrics to Google Cloud Monitoring")
+
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/executorqueue/metrics/gcp/reporter.go:30]: Could be spammy
```suggestion
```

Comment (efritz) [reply enterprise/cmd/frontend/internal/executorqueue/metrics/gcp/reporter.go:30]: This is in the constructor.

File: enterprise/cmd/frontend/internal/executorqueue/metrics/aws/reporter.go (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/frontend/internal/executorqueue/metrics/aws/reporter.go:26]: Yeah I'm updating this PR with auto similar to how we do it for S3 LSIF buckets.

**Review by efritz — APPROVED** [review]

**Pull request merged**